optuna/samplers/_tpe/sampler.py

```diff
@@ -514,12 +516,17 @@ def _build_parzen_estimator(
             weights_below = _calculate_weights_below_for_multi_objective(
                 study, trials, self._constraints_func
             )[param_mask_below]
-            mpe = _ParzenEstimator(
+            mpe = self._parzen_estimator_cls(
                 observations, search_space, self._parzen_estimator_parameters, weights_below
             )
         else:
-            mpe = _ParzenEstimator(observations, search_space, self._parzen_estimator_parameters)
+            mpe = self._parzen_estimator_cls(
+                observations, search_space, self._parzen_estimator_parameters
+            )
 
+        assert isinstance(
+            mpe, _ParzenEstimator
+        ), "_parzen_estimator_cls must override _ParzenEstimator"
```

Comment:
I think `RuntimeError` or `UserWarning` would be better here since `assert` is expected not to fail at runtime.